python/ray/data/_internal/stream_split_dataset_iterator.py

```diff
@@ -115,6 +117,18 @@ def gen_blocks() -> Iterator[ObjectRef[Block]]:
             shuffle_seed=local_shuffle_seed,
         )
 
+    def iter_rows(self, *, prefetch_blocks: int = 0) -> Iterator[Union[T, TableRow]]:
+        # During row-based ops, we also choose a batch format that lines up with the
+        # current dataset format in order to eliminate unnecessary copies and type
+        # conversions.
+        batch_format = _get_batch_format(self._base_dataset)
+        for batch in self.iter_batches(
```

Comment:
This is basically the same as the other impl. Could we make this not an abstract method, and delete the impl in this file?

Reply:
(move the common impl to dataset_iterator.py, and then delete the methods in stream_split_dataset_iterator.py and dataset_iterator_impl.py)

Reply:
Ok, I put it in this way because `PipelinedDatasetIterator.iter_rows` has its own impl. So technically these 3 subclasses not sharing same `iter_rows` impl. But I can put it inside `DatasetIterator.iter_rows`.

Reply:
I think that one can also use the same impl...

Reply:
(We could make get_batch_format() an abstract method instead, like schema)

Reply:
Actually I found [`PipelinedDatasetIterator.iter_batches()` iterate next dataset internally](https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/pipelined_dataset_iterator.py#L42), I don't want to further complicate `PipelinedDatasetIterator.iter_batches` interface. How about we leave the implenmentation as it is for now?

Reply:
To be clear, this is my proposal, and I believe we can remove most of the other code after this.

```
class DatasetIterator:
 def iter_rows(...):
   target_format = self._best_batch_format()
   for batch in self.iter_batches(target_format):
        ...
           yield row
           
 def _best_batch_format():
    return "default"   # can override to customize in DatasetIteratorImpl
```

Reply:
@ericl - ah got it. Updated.